### ISSUE TITLE
`modernize-use-emplace` not reported when braced initializer is being used

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#55870]
```cpp
#include <string>
#include <vector>

struct A {
    std::string s;
};

void f1() {
    std::vector<A> v;
    v.push_back(A()); // warning
}

void f2() {
    std::vector<A> v;
    v.push_back(A{}); // no warning
}

void f3() {
    std::vector<A> v;
    v.push_back(A{""}); // no warning
}

void f4() {
    std::vector<A> v;
    v.push_back({}); // no warning
}

void f5() {
    std::vector<A> v;
    v.push_back({""}); // no warning
}
```

https://godbolt.org/z/o18j88G7d

**Comment by firewave** [comment]
This also prevents the `unnecessary temporary object` version of this check being detected:

```cpp
#include <string>
#include <vector>

struct A {
    std::string s;
};

void f1() {
    std::vector<A> v;
    v.emplace_back(A()); // warning
}

void f2() {
    std::vector<A> v;
    v.emplace_back(A{}); // no warning
}

void f3() {
    std::vector<A> v;
    v.emplace_back(A{""}); // no warning
}
```
https://godbolt.org/z/GPYdjf8E7

**Comment by PatriotRossii** [comment]
I'll work on it

**Comment by BigPeet** [comment]
Hi, I would like to take a look at this. Is this still "open"? Should this be solved in tandem with #55869?

**Comment by BigPeet** [comment]
It seems like I'm close to a solution. But the fix seems to be broken.
I will have to check if I introduced that issue or if it was present before (and then try to fix it).

**Comment by BigPeet** [comment]
I've created a revision on Phabricator: https://reviews.llvm.org/D135405.
I hope I did everything correct, but if there is something wrong with the submission (code- or process-wise), please let me know.
